### ISSUE TITLE
Why does react-dates need @babel/runtime and react-with-direction?

### ISSUE THREAD

**Issue by daviddelusenet** [airbnb/react-dates#1858]
**react-dates version**
react-dates@21.2.1

**Describe the bug**
The documentation states that if you run the following command:

```shell
(
  export PKG=react-dates;
  npm info "$PKG" peerDependencies --json | command sed 's/[\{\},]//g ; s/: /@/g; s/ *//g' | xargs npm install --save "$PKG"
)
```

It will result in the following:

```shell
npm install --save react-dates moment@>=#.## react@>=#.## react-dom@>=#.##
```

However, it also installs `react-with-direction` and `@babel/runtime` as dependencies. Why does this happen? Are these dependencies even needed?

**Comment by ljharb** [comment]
Yes, they’re all listed peer deps, which is why the command is helpful. The readme example just hasn’t been updated with the new ones.

**Comment by daviddelusenet** [comment]
@ljharb thanks for response.

So I'm building a React UI kit and one of the components in there is using the `SingleDatePicker` component from the `react-dates` library.

For now I've only specified `react-dates` and `moment` as peer dependencies of my UI kit. I'm using an instance of a Create React App to test if my UI kit is working.

While I only install `react-dates` and `moment` in my CRA the date picker component I build is still working. Is this because those dependencies only get used in certain cases or?

**Comment by ljharb** [comment]
If `npm ls` returns nonzero, then your entire dep graph is invalid and can't be relied upon to work properly. Everything that's a peer dep *must* be installed, always, full stop.

Certainly there may be code paths where some of those deps aren't used; they also might be accidentally installed already due to npm's dep hoisting - but you still should always explicitly specify them.

**Comment by ljharb** [comment]
Note that if your package is published, all those peer deps should be *both* peer and dev deps, to pass on the requirement to consumers.

**Comment by daviddelusenet** [comment]
Thanks, will add them as peer dependencies!

I think `@babel/runtime` is already installed as a CRA dependency and `react-with-direction` will probably only be used in certain use cases which I didn't encounter yet.

**Comment by ljharb** [comment]
It still needs to always be installed, no matter what.
